core/serial_command.py

```diff
@@ -81,12 +81,18 @@ class SerialCommand(object):
     """Contains a serial command object.
     """
 
-    def __init__(self, port_name=None):
+    def __init__(self, port_name=None, port_timeout=None):
```

Comment:
Are you sure you want port_timeout = None? That is a real, possible timeout setting in pyserial, and means the function will block until all the expected data has been received. With the current logic (where we always ask to read 100 bits) this will hang forever.